### ISSUE TITLE
Multiple MathML blocks allowed?

### ISSUE THREAD

**Issue by kerimoyle** [cellml/libcellml#556]
Related to https://github.com/cellml/cellml-specification/issues/27

If (and I'm not sure they are) multiple `<math>` blocks within a single component are valid CellML, then we are not currently reading them in libcellml.  If they're not, then we need to say so in the spec.  This test fails:

```
TEST(Validator, validMathInMultipleMathMlBlocks)
{
    const std::string math =
        "<math xmlns:cellml=\"http://www.cellml.org/cellml/2.0#\" xmlns=\"http://www.w3.org/1998/Math/MathML\">\n"
        "  <apply>\n"
        "    <eq/>\n"
        "    <ci>A</ci>\n"
        "    <apply>\n"
        "      <plus/>\n"
        "      <cn cellml:units=\"dimensionless\">1</cn>\n"
        "    </apply>\n"
        "  </apply>\n"
        "</math>\n"
        "<math xmlns:cellml=\"http://www.cellml.org/cellml/2.0#\" xmlns=\"http://www.w3.org/1998/Math/MathML\">\n"
        "  <apply>\n"
        "    <eq/>\n"
        "    <ci>B</ci>\n"
        "    <apply>\n"
        "      <plus/>\n"
        "      <cn cellml:units=\"dimensionless\">2</cn>\n"
        "    </apply>\n"
        "  </apply>\n"
        "</math>\n";

    libcellml::ValidatorPtr v = libcellml::Validator::create();
    libcellml::ModelPtr m = libcellml::Model::create();
    libcellml::ComponentPtr c = libcellml::Component::create();
    libcellml::VariablePtr v1 = libcellml::Variable::create();
    libcellml::VariablePtr v2 = libcellml::Variable::create();

    m->setName("modelName");
    c->setName("componentName");
    v1->setName("A");
    v2->setName("B");
    v1->setUnits("dimensionless");
    v2->setUnits("dimensionless");

    c->addVariable(v1);
    c->addVariable(v2);
    c->setMath(math);
    m->addComponent(c);

    v->validateModel(m);
    EXPECT_EQ(size_t(0), v->errorCount());
    printErrors(v);
}
```

Reporting errors:
```
[ RUN      ] Validator.validMathInMultipleMathMlBlocks
/Users/kmoy001/libcellml/libcellml-src/tests/validator/validator.cpp:472: Failure
Expected equality of these values:
  size_t(0)
    Which is: 0
  v->errorCount()
    Which is: 2
Error   1: LibXml2 error: Extra content at the end of the document.
Error   2: Could not get a valid XML root node from the math on component 'componentName'.
[  FAILED  ] Validator.validMathInMultipleMathMlBlocks (0 ms)
```

**Comment by hsorby** [comment]
I think we have been talking about saying that the math of a component must be a valid sub-document.  However I can find no written evidence of this anywhere.  It is certainly not in the spec.

**Comment by kerimoyle** [comment]
This also applies to the `test_value` and `reset_value` sections, where the spec says explicitly that each must contain "at least one mathml child" ... meaning (I presume) that more than one is allowed!  Those tests fail too - see #557

**Comment by kerimoyle** [comment]
Closed by #557
